File: app/src/main/java/de/eloc/eloc_control_panel/ng2/models/PreferredFontSize.java
package de.eloc.eloc_control_panel.ng2.models;

public enum PreferredFontSize {
    small(-1),
    medium(0),
    large(1);

    private final int size;

    PreferredFontSize(int size) {
        this.size = size;
    }

    public int getSize() {
        return size;
    }

    public static PreferredFontSize fromInt(int size) {
        if (size < 0) {
            return small;
        } else if (size > 0) {
            return large;
        } else {
            return medium;
        }
    }
}

File: app/src/main/java/de/eloc/eloc_control_panel/ng3/data/PreferredFontSize.kt
package de.eloc.eloc_control_panel.ng3.data

enum class PreferredFontSize(val code: Int) {
    Small(-1),
    Medium(0),
    Large(1);

    companion object {
        fun parse(code: Int) = if (code < 0) {
            Small
        } else if (code > 0) {
            Large
        } else {
            Medium
        }
    }
}